### ISSUE TITLE
TOMAS failure to integrate in v14.4.3

### ISSUE THREAD

**Issue by BettyCroft** [geoschem/gchp#457]
### Your name

Betty Croft

### Your affiliation

Washington University in St. Louis

### What happened? What did you expect to happen?

GCHP fails to integrate with TOMAS in v.14.4.3. However integration of GCHP with TOMAS is successful in v.14.4.2. 

### What are the steps to reproduce the bug?

Configure the build with -DTOMAS=y -DTOMAS_BINS=15 and compile, and launch a simulation. Simulation aborts prior to time-stepping. Related issue also noted with simulation slowness > 10-fold when using TOMAS in classic configuration of GEOS-Chem. Could a parallelization issue for TOMAS have arisen since v14.4.3?

### What GCHP version were you using?

14.4.3

### What environment were you running GCHP on?

Local cluster

### What compiler and version were you using?

gcc12.2.0

### What MPI library and version were you using?

IntelMPI

### Will you be addressing this bug yourself?

No

### Additional information

Happy to assist with testing any advice on fixes. Thanks.

**Comment by BettyCroft** [comment]
Hi @yantosca - I think that we can resolve this issue as related to GCHP. I've been able to successfully compile and run GCHP-TOMAS in v14.5.0. The 15 bin GCHP-TOMAS runs without modifications - and I've determined that the maximum number of advected species was exceeded for 40-bin GCHP-TOMAS and causing job failure. This issue appears to be  resolved with revision to both Interfaces/Chem_GridCompMod.F90 and Headers/Input_opt_mod.F90 to increase the maximum number of advected species to 700. The slowness for GC-Classic remains unresolved at this time - so would appreciate keeping that issue open [ ](https://github.com/geoschem/geos-chem/issues/2556#issuecomment-2455838042) Thanks!
